src/Kdyby/Validator/ConstraintValidatorFactory.php

```diff
@@ -47,23 +49,16 @@ public function __construct(Nette\DI\Container $serviceLocator)
 	 */
 	public function getInstance(Constraint $constraint)
 	{
-		$className = $constraint->validatedBy();
+		$name = $constraint->validatedBy();
 
-		// Workaround for https://github.com/symfony/symfony/pull/16166.
-		if ($className === 'validator.expression') {
-			$className = 'Symfony\Component\Validator\Constraints\ExpressionValidator';
+		if (!isset($this->validators[$name])) {
+			trigger_error(sprintf('Please register all you constraint validators as DI services, namely %s.', $name), E_USER_DEPRECATED);
+			$this->validators[$name] = $this->serviceLocator->createInstance($name);
```

Comment:
Yeah, that's one of the options I suggested earlier but it's BC break because validators with non-optional parameters in ctor won't work at all unless registered in DIC. I'm fine with that though.